### PR TITLE
koji_import: re-write pull specifications for Crane as needed

### PR DESCRIPTION
**Pull request by twaugh**
When Pulp and Koji integration are both in use, Koji metadata created by worker builds has incorrect pullspecs in the extra.docker.repositories field for the docker-image output. This is because the worker build only knows about the 'sync' Docker registry (used for syncing images into Pulp).

In this situation, koji_import needs to re-write these pull specifications so they use Pulp's Crane instead of the Docker registry.

Signed-off-by: Tim Waugh <twaugh@redhat.com>

### CODE REVIEW

File: atomic_reactor/plugins/exit_koji_import.py [hunk 1]
```diff
@@ -99,14 +102,34 @@ def get_output(self, worker_metadatas):
...
+
         for platform in worker_metadatas:
             for instance in worker_metadatas[platform]['output']:
                 instance['buildroot_id'] = '{}-{}'.format(platform, instance['buildroot_id'])
 
+                if instance['type'] != 'docker-image':
+                    continue
+
                 # update image ID with pulp_pull results
-                if 'extra' in instance and platform == "x86_64" and has_pulp_pull:
```
Comment (vrutkovs) [inline atomic_reactor/plugins/exit_koji_import.py]: `'extra' in instance` check is required - as logs are being listed here

File: atomic_reactor/plugins/exit_koji_import.py [hunk 2]
```diff
@@ -99,14 +102,34 @@ def get_output(self, worker_metadatas):
...
+                if platform == "x86_64" and has_pulp_pull:
                     image_id, _ = self.workflow.postbuild_results[PLUGIN_PULP_PULL_KEY]
                     instance['extra']['docker']['id'] = image_id
+
+                # update repositories to point to Crane
+                if crane_registry:
+                    pulp_pullspecs = []
+                    for pullspec in instance['extra']['docker']['repositories']:
```
Comment (vrutkovs) [inline atomic_reactor/plugins/exit_koji_import.py]: Same here: `instance` without `'extra'` key should be filtered out

File: atomic_reactor/plugins/exit_koji_import.py (hunk 1 above)
Comment (twaugh) [inline atomic_reactor/plugins/exit_koji_import.py]: But see a few lines above where I filter on 'type'.

File: atomic_reactor/plugins/exit_koji_import.py (hunk 2 above)
Comment (twaugh) [inline atomic_reactor/plugins/exit_koji_import.py]: See a few lines above where I filter on 'type'.

File: atomic_reactor/plugins/exit_koji_import.py (hunk 1 above)
Comment (vrutkovs) [inline atomic_reactor/plugins/exit_koji_import.py]: Oh, its a bit subtle, but yes, would work

File: atomic_reactor/plugins/post_koji_upload.py [hunk 3]
```diff
@@ -473,7 +473,7 @@ def add_log_type(output, arch):
 
         digests = self.get_digests()
         repositories = self.get_repositories(digests)
-        tags = set(image.tag for image in self.workflow.tag_conf.primary_images)
+        tags = set(image.tag for image in self.workflow.tag_conf.images)
```
Comment (lcarva) [inline atomic_reactor/plugins/post_koji_upload.py]: Why is this needed now?

Comment (twaugh) [inline atomic_reactor/plugins/post_koji_upload.py]: There are literally no primary tags for worker builds any more, so the metadata included "tags: []". Hmm, this should go in a separate commit I guess.

Comment (twaugh) [inline atomic_reactor/plugins/post_koji_upload.py]: Separated into its own commit.

**Comment by lcarva** [comment]
Bugfix to unreleased code - not updating release notes.
